src/behaviors/anchoredPosition.ts

```diff
@@ -0,0 +1,376 @@
+export type AnchoredPositionAlign = 'first' | 'center' | 'last'
+
+// When prettier supports template literal types...
+// export type AnchorSide = `${'inside' | 'outside'}-${'top' | 'bottom' | 'right' | 'left'}` | 'inside-center'
+export type AnchorSide =
+  | 'inside-top'
+  | 'inside-bottom'
+  | 'inside-left'
+  | 'inside-right'
+  | 'inside-center'
+  | 'outside-top'
+  | 'outside-bottom'
+  | 'outside-left'
+  | 'outside-right'
+
+/**
+ * Settings that customize how a floating element is positioned
+ * with respect to an anchor element.
+ */
+export interface PositionSettings {
+  /**
+   * Sets the side of the anchor element that the floating element should be
+   * pinned to. This side is given by a string starting with either "inside" or
+   * "outside", followed by a hyphen, followed by either "top", "right", "bottom",
+   * or "left". Additionally, "inside-center" is an allowed value.
+   *
+   * The first part of this string, "inside" or "outside", determines whether the
+   * floating element should be attempted to be placed "inside" the anchor element
+   * or "outside" of it. Using "inside" is useful for making it appear that the
+   * anchor _contains_ the floating element, and it can be used for implementing a
+   * dialog that is centered on the screen. The "outside" value is more common and
+   * can be used for tooltips, popovers, menus, etc.
+   *
+   * The second part of this string determines the _edge_ on the anchor element that
+   * the floating element will be anchored to. If side is "inside-center", then
+   * the floating element will be centered in the X-direction (while align is used
+   * to position it in the Y-direction).
+   * Note: "outside-center" is _not_ a valid value for this property.
+   */
+  side: AnchorSide
+
+  /**
+   * Determines how the floating element should align with the anchor element. If
+   * set to "first", the floating element's first edge (top or left) will align
+   * with the anchor element's first edge. If set to "center", the floating
+   * element will be centered along the axis of the anchor edge. If set to "last",
+   * the floating element's last edge will align with the anchor element's last edge.
+   */
+  align: AnchoredPositionAlign
```

Comment:
I think we chatted about changing this to `start` and `end` to be consistent with the values of the same sort in CSS